### PR TITLE
Document RSpec and RuboCop artifact formats

### PR DESCRIPTION
**Pull request by edigaryev**
See cirruslabs/cirrus-ci-annotations#10 and cirruslabs/cirrus-ci-annotations#11.

### CODE REVIEW

File: docs/examples.md [hunk 1]
```diff
@@ -553,6 +553,34 @@ rspec_task:
     Which will create four shards that will theoretically **run tests 4x faster** by equaly splitting all tests between 
     these four shards.
 
+### RSpec and RuboCop Annotations
+
+Cirrus CI natively supports [RSpec](https://rspec.info/) and [RuboCop](https://rubocop.org/) machine-parsable JSON reports.
+
+To get Behaviour-Driven Development test annotations, simply generate a `rspec` artifact from your linter task:
+
+```yaml
```
Comment (RDIL) [inline docs/examples.md:562]: Please add a container instruction for both examples with Ruby

File: docs/examples.md [hunk 2]
```diff
@@ -553,6 +553,34 @@ rspec_task:
     Which will create four shards that will theoretically **run tests 4x faster** by equaly splitting all tests between 
     these four shards.
 
+### RSpec and RuboCop Annotations
+
+Cirrus CI natively supports [RSpec](https://rspec.info/) and [RuboCop](https://rubocop.org/) machine-parsable JSON reports.
+
+To get Behaviour-Driven Development test annotations, simply generate a `rspec` artifact from your linter task:
```
Comment (RDIL) [inline docs/examples.md:560]: Please use en-US spelling, and this sentence can be improved a bit:
```suggestion
To get behavior-driven test annotations, simply generate a `rspec` artifact from your lint task:
```

File: docs/examples.md (hunk 1 above)
Comment (fkorotkov) [reply docs/examples.md:562]: I'd also recommend adding `name: RSpec` and `name: RuboCop`

File: docs/examples.md [hunk 3]
```diff
@@ -553,6 +553,34 @@ rspec_task:
...
+      path: "rspec.json"
+      type: text/json
+      format: rspec
+```
+
+Generate a `rubocop` artifact to quickly gain context for linter/formatter annotations:
+
+```yaml
+task:
+  script: rubocop --format json --out rubocop.json
```
Comment (fkorotkov) [inline docs/examples.md:576]: Does `rubocop` is a part of the standard `ruby` Docker image?

Comment (RDIL) [reply docs/examples.md:576]: No, it needs to either be installed with RubyGems or Bundler

File: docs/examples.md (hunk 2 above)
Comment (edigaryev) [reply docs/examples.md]: Thanks!

File: docs/examples.md (hunk 1 above)
Comment (edigaryev) [reply docs/examples.md:562]: See ac443369329753292143b6e7153bc40574661412.

File: docs/examples.md (hunk 3 above)
Comment (edigaryev) [reply docs/examples.md]: See 35c30329f54cb8b8711557f2cf2707b5ff9d4c04.

File: docs/examples.md (hunk 1 above)
Comment (fkorotkov) [reply docs/examples.md:562]: Let's also add the annotations to the main example in the `Ruby` section (after `rspec_script: bundle exec rspec`).

**Comment by fkorotkov** [comment]
I also think a "true Ruby" way will be to use `bundle`. See [publicly available examples of `.cirrus.yml` with RSpec](https://github.com/search?l=&q=RSpec+filename%3A.cirrus.yml&type=Code). Everyone is using bundle 🤔

**Comment by edigaryev** [comment]
>Let's also add the annotations to the main example in the Ruby section (after rspec_script: bundle exec rspec).

See https://github.com/cirruslabs/cirrus-ci-docs/pull/668/commits/2137c425fdc31da40cc5862ca8d463f8e80eb3fd.

>I also think a "true Ruby" way will be to use bundle.

See https://github.com/cirruslabs/cirrus-ci-docs/pull/668/commits/a16625bb24ae613f019b450c21d712ef0c15b380.

Also, I've added a caching afterwards (https://github.com/cirruslabs/cirrus-ci-docs/pull/668/commits/f527ca33f580f0abbdabe2eff17ed3577102605b), it looks somewhat boilerplate-ish (considering these two examples come one after another), so this change can be reverted separately if needed.

**Pull request merged**